vectorstores/pgvector/engine.go

```diff
@@ -0,0 +1,141 @@
+package pgvector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/oauth2/v2"
+	"google.golang.org/api/option"
+)
+
+const Port = 5432 // default PostgreSQL port
+
+type ServiceAccountRetriever interface {
+	GetServiceAccountEmail(ctx context.Context) (string, error)
+}
+type DefaultServiceAccountRetriever struct{}
+
+func (d *DefaultServiceAccountRetriever) GetServiceAccountEmail(ctx context.Context) (string, error) {
+	return getServiceAccountEmail(ctx)
+}
+
+type Column struct {
+	Name     string
+	DataType string
+	Nullable bool
+}
+
+func NewColumn(name, dataType string, nullable bool) (*Column, error) {
+	if name == "" {
+		return nil, errors.New("name should be provided")
+	}
+	if dataType == "" {
+		return nil, errors.New("dataType should be provided")
+	}
+	return &Column{Name: name, DataType: dataType, Nullable: nullable}, nil
+}
```

Comment:
It is unusual to both export fields and have a `NewColumn` setter. Is there a reason for that choice here?

Reply:
Sorry, I just wrote Column and NewColumn as quick designs. I will remove them until they are needed on future methods, and will create them the correct way.